### PR TITLE
Kubernetes controllers

### PR DESCRIPTION
**Pull request by drozdnikita**
# Выполнено ДЗ № 2

 - [x] Основное ДЗ
 - [x] Задание со *

## В процессе сделано:
 - ReplicaSet with replicas set to 3. Got to know hot to re-deploy using rs (via kubectl pod delete, because rs doesn't verify PODs against templates it just makes sure that the number of running PODs equals the desired number of PODs)
 - Deployments with replicas set to 3 + played around with maxSurge & maxUnavailable parameters in order to implement blue-green like deployment and reverse as follows:
 1. blue/green
- maxSurge: 3
- maxUnavailable: 0
2. remove 1 old / create 1 new
- maxSurge: 0
- maxUnavailable: 1

- for some reason didn't manage to deploy node-exporter via DaemonSet:
1. created cluster via kind
`NAME                  STATUS   ROLES    AGE   VERSION`
`kind-control-plane    Ready    master   75m   v1.19.1`
`kind-control-plane2   Ready    master   74m   v1.19.1`
`kind-control-plane3   Ready    master   73m   v1.19.1`
`kind-worker           Ready    <none>   73m   v1.19.1`
`kind-worker2          Ready    <none>   73m   v1.19.1`
`kind-worker3          Ready    <none>   73m   v1.19.1`

2. then tried to apply the attached node-exporter manifest as follows
`kubectl apply -f node-exporter-daemonset.yaml`
3. got the following result
`kubectl get daemonsets -n monitoring` 
`NAME            DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE`
`node-exporter   0         0         0       0            0           kubernetes.io/os=linux   56m`
 
see manifests in the repository

## Как запустить проект:
apply either of manifest, for example
`kubectl apply -f frontend-replicaset.yaml`
`kubectl apply -f frontend-deployment.yaml`
`...`
`kubectl apply -f payment-deployment-bg.yaml`
`kubectl apply -f payment-deployment-reverse.yaml`
`...`

## PR checklist:
 - [x] Выставлен label с темой домашнего задания

### CODE REVIEW

**Review by Jasstkn — CHANGES_REQUESTED** [review]
Добрый день.
Во время прохождения тестов возникла ошибка:
> TestKubernetesPaymentDeployment 2021-01-11T17:55:00Z command.go:158: error: the path "/home/mariak/hw/drozdnikita_platform/kubernetes-controllers/paymentservice-deployment.yaml" does not exist

**Comment by drozdnikita** [comment]
> Добрый день.
> Во время прохождения тестов возникла ошибка:
> 
> > TestKubernetesPaymentDeployment 2021-01-11T17:55:00Z command.go:158: error: the path "/home/mariak/hw/drozdnikita_platform/kubernetes-controllers/paymentservice-deployment.yaml" does not exist

Добрый день, исправил имена файлов

File: kubernetes-controllers/node-exporter-daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      labels:
+        app.kubernetes.io/component: exporter
+        app.kubernetes.io/name: node-exporter
+        app.kubernetes.io/part-of: kube-prometheus
+        app.kubernetes.io/version: 1.0.1
+    spec:
+      tolerations:
+      # this toleration is to have the daemonset runnable on master nodes
+      # remove it if your masters can't run pods
+      - key: node-role.kubernetes.io/master
```
Comment (Jasstkn) [inline kubernetes-controllers/node-exporter-daemonset.yaml:28]: Этот вариант тоже верный, поды запустятся на всех нодах:
```
 tolerations:
   - operator: Exists
```

**Review by Jasstkn — APPROVED** [review]
Теперь все верно!

File: kubernetes-controllers/node-exporter-daemonset.yaml (hunk 1 above)
Comment (drozdnikita) [reply kubernetes-controllers/node-exporter-daemonset.yaml:28]: принято, спасибо!

**Pull request merged**